File: java/udf/WordCountReducer.java
package udf;

import java.util.List;

import mr.Pair;
import mr.Reducer;

public class WordCountReducer implements Reducer<String, Integer, String, Integer> {

    @Override
    public Pair<String, Integer> reduce(String key, List<Integer> values) {
        return new Pair<String,Integer>(key, values.stream().mapToInt(e -> e).sum());
    }
}

File: golang/udf/wordcount_reducer.go
package main

import "mr/mr"

var Reduce mr.Reduce = func(key any, values []any) *mr.Pair {
	count := 0
	for _, value := range values {
		count += value.(int)
	}
	return &mr.Pair{Key: key, Value: count}
}
